### PR TITLE
Cortex-M: Fix pendsv induced crashes

### PR DESCRIPTION
**Pull request by agkaminski**
## Description

Hopefully fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1017 and https://github.com/phoenix-rtos/phoenix-rtos-project/issues/885 too

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/cpu.c [hunk 1]
```diff
@@ -162,6 +159,7 @@ int hal_cpuPushSignal(void *kstack, void (*handler)(void), cpu_context_t *signal
 
 	signalCtx->psp -= sizeof(cpu_context_t);
 	signalCtx->hwctx.pc = (u32)handler;
+	signalCtx->hwctx.psr = 0x61000000;
```
Comment (lukileczo) [inline hal/armv7m/cpu.c:162]: Why do we need this value here? I think a comment also would be nice

**Review by lukileczo — COMMENTED** [review]
One remark only, otherwise LGTM

**Comment by lukileczo** [comment]
And I'm not sure if it's worth including the commits changing the pendsv handler which gets removed altogether in the next commit

**Comment by agkaminski** [comment]
> And I'm not sure if it's worth including the commits changing the pendsv handler which gets removed altogether in the next commit

I wanted to keep bug fix in the history, just in case

**Review by lukileczo — APPROVED** [review]

**Pull request merged**